tslearn/shapelets.py

```diff
@@ -407,39 +410,35 @@ def fit(self, X, y):
         self.model_ = None
         self.transformer_model_ = None
         self.locator_model_ = None
-        self.categorical_y_ = False
-        self.label_binarizer_ = None
         self.d_ = d
 
-        if y.ndim == 1 or y.shape[1] == 1:
-            self.label_binarizer_ = LabelBinarizer().fit(y)
-            y_ = self.label_binarizer_.transform(y)
-            self.classes_ = self.label_binarizer_.classes_
+        self.classes_ = unique_labels(y)
```

Comment:
Perhaps we could create a private function for this to reduce the complexity of the `fit` method (`_process_labels` or something?)

Reply:
yep, why not